### PR TITLE
Add ability for models to have string Ids

### PR DESCRIPTION
**Pull request by lewisf2001uk**
Add ability for models to have string Ids (UUID).
Before this fix $idsOnly and $idsAttributes were returning empty arrays.

### CODE REVIEW

File: src/Relations/BelongsToManyCustom.php [hunk 1]
```diff
@@ -84,7 +84,7 @@ private function getIdsWithAttributes($id, $attributes = [])
                     $ids[$attributesArray] = $attributes;
                 }
             }
-        } elseif (is_int($id)) {
```
Comment (fico7489) [inline src/Relations/BelongsToManyCustom.php:87]: 1.is it better to add 

`...} elseif (is_int($id)  ||  is_string($id)) {...`

?

File: tests/PivotEventTraitTest.php [hunk 2]
```diff
@@ -33,22 +37,35 @@ public function setUp()
     private function startListening()
```
Comment (fico7489) [inline tests/PivotEventTraitTest.php:37]: Maybe we can rename function to startListeningAndGetUser()  or something like that?

File: src/Relations/BelongsToManyCustom.php (hunk 1 above)
Comment (lewisf2001uk) [reply src/Relations/BelongsToManyCustom.php:87]: Personally I would say not, as you will want a default option.
If I'm honest my coding style is more return early. So I would have if's without else's and return in the if's.
At the end of the day it is your package however.

File: tests/PivotEventTraitTest.php (hunk 2 above)
Comment (lewisf2001uk) [reply tests/PivotEventTraitTest.php:37]: I would suggest against that too keep the code more SOLID, (S - Single-responsiblity principle).
Also "startListening" is very verbose about what it is doing and `$user = User::first()` is very obvious knowing what it doing.
When I went to do the pull request I first was going about it slightly differently and the fact that `return User::first()` was in the function tripped me up slightly.

File: src/Relations/BelongsToManyCustom.php (hunk 1 above)
Comment (fico7489) [reply src/Relations/BelongsToManyCustom.php:87]: @lewisf2001uk  Actually if someone by accident passes instance that is not derived from Illuminate\Database\Eloquent\Model
he will get "Illegal offset type" error.

File: tests/PivotEventTraitTest.php (hunk 2 above)
Comment (fico7489) [reply tests/PivotEventTraitTest.php:37]: SOLID vs DRY

File: src/Relations/BelongsToManyCustom.php (hunk 1 above)
Comment (lewisf2001uk) [reply src/Relations/BelongsToManyCustom.php:87]: Yeah, I was just looking at how Laravel / Eloquent deals with it in the `parseIds` function of `InteractsWithPivotTable` trait, they don't do the check of string / int.
I suppose passing back an empty array is better than receiving an exception.

File: tests/PivotEventTraitTest.php (hunk 2 above)
Comment (lewisf2001uk) [reply tests/PivotEventTraitTest.php:37]: There are cases where `$this->startListening()` is called multiple times in a a test but the `return User::first()` is only used the first time it is called.
If you want to be more DRY, wouldn't it be better to have `$this->firstUser = User::find(1)` or  `$this->firstUser = User::create(['name' => 'example@example.com']);` within the `setUp` function?

**Comment by lewisf2001uk** [comment]
Let me know what changes you want done and I'll add another commit to this pull request.

File: src/Relations/BelongsToManyCustom.php (hunk 1 above)
Comment (fico7489) [reply src/Relations/BelongsToManyCustom.php:87]: If I understood correctly, you agree with me? can you then push that commit so I can merge PR?
PS. I will leave tests as they are in this PR.

**Comment by lewisf2001uk** [comment]
@fico7489 Apologies it took so long to get this last commit, I wasn't available last night.

**Pull request merged**